File: hcx-pipeline-jobs/preauth-job/src/main/scala/org/swasth/dp/preauth/functions/PreauthProcessFunction.scala
package org.swasth.dp.preauth.functions

import org.apache.flink.api.common.typeinfo.TypeInformation
import org.slf4j.LoggerFactory
import org.swasth.dp.core.function.{BaseDispatcherFunction, ValidationResult}
import org.swasth.dp.preauth.task.PreauthConfig

import java.util

class PreauthProcessFunction(config: PreauthConfig)(implicit val stringTypeInfo: TypeInformation[String])
  extends BaseDispatcherFunction(config) {

  private[this] val logger = LoggerFactory.getLogger(classOf[PreauthProcessFunction])

  override def validate(event: util.Map[String, AnyRef]): ValidationResult = {
    // TODO: Add domain specific validations
    ValidationResult(status = true, None)
  }

}

File: hcx-pipeline-jobs/preauth-job/src/main/java/org/swasth/dp/preauth/functions/PreauthProcessFunction.java
package org.swasth.dp.preauth.functions;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.swasth.dp.core.function.BaseDispatcherFunction;
import org.swasth.dp.core.function.ValidationResult;
import org.swasth.dp.preauth.task.PreauthConfig;

import java.util.Map;

public class PreauthProcessFunction extends BaseDispatcherFunction {

  private Logger logger = LoggerFactory.getLogger(PreauthProcessFunction.class);
  private PreauthConfig config;

  public PreauthProcessFunction(PreauthConfig config) {
    super(config);
    this.config = config;
  }

  @Override
  public ValidationResult validate(Map<String, Object> event) {
    // TODO: Add domain specific validations
    return new ValidationResult(true, null);
  }

}



